Author: Andreas Beckmann <anbe@debian.org>
Description: NET_ADDR_RANDOM was added in c1f79426e2df5 (v2.6.36-rc1~571^2~104)

--- a/src/r8168_n.c
+++ b/src/r8168_n.c
@@ -25299,7 +25299,9 @@ rtl8168_get_mac_address(struct net_devic
                 netif_err(tp, probe, dev, "Invalid ether addr %pM\n",
                           mac_addr);
                 eth_random_addr(mac_addr);
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,36)
                 dev->addr_assign_type = NET_ADDR_RANDOM;
+#endif
                 netif_info(tp, probe, dev, "Random ether addr %pM\n",
                            mac_addr);
                 tp->random_mac = 1;
