Author:  Jan Schwarzrock <pssgcsim@psb1.org>
Description: fix building for Linux 4.11
Bug-Debian: 864867

diff --git a/src/r8168_n.c b/src/r8168_n.c
index 3c5a7e7..4086291 100755
--- a/src/r8168_n.c
+++ b/src/r8168_n.c
@@ -25731,8 +25731,9 @@ process_pkt:

                         if (rtl8168_rx_vlan_skb(tp, desc, skb) < 0)
                                 rtl8168_rx_skb(tp, skb);
-
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,11,0)
                         dev->last_rx = jiffies;
+#endif
                         RTLDEV->stats.rx_bytes += pkt_size;
                         RTLDEV->stats.rx_packets++;
                 }
