Author: Andreas Beckmann <anbe@debian.org>
Description: adapt to Linux 5.6 changes

--- a/src/r8168_n.c
+++ b/src/r8168_n.c
@@ -456,7 +456,11 @@ static void rtl8168_hw_config(struct net
 static void rtl8168_hw_start(struct net_device *dev);
 static int rtl8168_close(struct net_device *dev);
 static void rtl8168_set_rx_mode(struct net_device *dev);
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,6,0)
 static void rtl8168_tx_timeout(struct net_device *dev);
+#else
+static void rtl8168_tx_timeout(struct net_device *dev, unsigned int txqueue);
+#endif
 static struct net_device_stats *rtl8168_get_stats(struct net_device *dev);
 static int rtl8168_rx_interrupt(struct net_device *, struct rtl8168_private *, napi_budget);
 static int rtl8168_change_mtu(struct net_device *dev, int new_mtu);
@@ -1616,6 +1620,14 @@ static int rtl8168_proc_open(struct inod
         return single_open(file, show, dev);
 }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,6,0)
+static const struct proc_ops rtl8168_proc_fops = {
+        .proc_open           = rtl8168_proc_open,
+        .proc_read           = seq_read,
+        .proc_lseek          = seq_lseek,
+        .proc_release        = single_release,
+};
+#else
 static const struct file_operations rtl8168_proc_fops = {
         .open           = rtl8168_proc_open,
         .read           = seq_read,
@@ -1623,6 +1635,7 @@ static const struct file_operations rtl8
         .release        = single_release,
 };
 #endif
+#endif
 
 /*
  * Table of proc files we need to create.
@@ -27852,8 +27865,13 @@ static void rtl8168_reset_task(struct wo
         }
 }
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,6,0)
 static void
 rtl8168_tx_timeout(struct net_device *dev)
+#else
+static void
+rtl8168_tx_timeout(struct net_device *dev, unsigned int txqueue)
+#endif
 {
         struct rtl8168_private *tp = netdev_priv(dev);
         unsigned long flags;
